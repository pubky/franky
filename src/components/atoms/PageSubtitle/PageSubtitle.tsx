import { ReactNode } from 'react';
import { cn } from '@/libs';

interface PageSubtitleProps extends React.HTMLAttributes<HTMLHeadingElement> {
  children?: ReactNode;
  as?: 'h2' | 'h5' | 'p';
  title?: string;
}

export function PageSubtitle({ as: Component = 'h2', className, children, title, ...props }: PageSubtitleProps) {
  return (
    <Component
      className={cn('text-xl lg:text-2xl text-muted-foreground font-light leading-normal', className)}
      {...props}
    >
      {title || children}
    </Component>
  );
}
