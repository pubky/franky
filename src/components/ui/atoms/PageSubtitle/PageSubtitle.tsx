import { ReactNode } from 'react';
import { cn } from '@/libs';

interface PageSubtitleProps {
  children: ReactNode;
  className?: string;
  as?: 'h2' | 'h5' | 'p';
}

export function PageSubtitle({ children, className, as: Component = 'h2' }: PageSubtitleProps) {
  return (
    <Component className={cn('text-xl lg:text-2xl text-muted-foreground font-light leading-normal', className)}>
      {children}
    </Component>
  );
}
