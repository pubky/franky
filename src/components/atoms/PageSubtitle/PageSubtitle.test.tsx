import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { PageSubtitle } from './PageSubtitle';

describe('PageSubtitle', () => {
  it('renders with default props', () => {
    render(<PageSubtitle>Test subtitle</PageSubtitle>);
    const pageSubtitle = screen.getByText('Test subtitle');
    expect(pageSubtitle).toBeInTheDocument();
  });
});

describe('PageSubtitle - Snapshots', () => {
  it('matches snapshot with default props', () => {
    const { container } = render(<PageSubtitle>Test subtitle</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for h2 element', () => {
    const { container } = render(<PageSubtitle as="h2">H2 subtitle</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for h5 element', () => {
    const { container } = render(<PageSubtitle as="h5">H5 subtitle</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for p element', () => {
    const { container } = render(<PageSubtitle as="p">P subtitle</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with custom className', () => {
    const { container } = render(<PageSubtitle className="custom-subtitle">Custom subtitle</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with title prop', () => {
    const { container } = render(<PageSubtitle title="Title prop subtitle" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with simple content', () => {
    const { container } = render(<PageSubtitle>Simple subtitle</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with complex children', () => {
    const { container } = render(
      <PageSubtitle>
        <span>Complex</span> <strong>subtitle</strong> content
      </PageSubtitle>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with id prop', () => {
    const { container } = render(<PageSubtitle id="subtitle-id">Subtitle with ID</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with data-testid prop', () => {
    const { container } = render(<PageSubtitle data-testid="page-subtitle">Subtitle with test ID</PageSubtitle>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
