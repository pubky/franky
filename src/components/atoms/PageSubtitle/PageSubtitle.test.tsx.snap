// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageSubtitle - Snapshots > matches snapshot with default props 1`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  Test subtitle
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for additional props 1`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
  id="subtitle-id"
>
  Subtitle with ID
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for additional props 2`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
  data-testid="page-subtitle"
>
  Subtitle with test ID
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different HTML elements 1`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  H2 subtitle
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different HTML elements 2`] = `
<h5
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  H5 subtitle
</h5>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different HTML elements 3`] = `
<p
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  P subtitle
</p>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different children types 1`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  Simple subtitle
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different children types 2`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  <span>
    Complex
  </span>
   
  <strong>
    subtitle
  </strong>
   content
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different configurations 1`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  Default subtitle
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different configurations 2`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal custom-subtitle"
>
  Custom subtitle
</h2>
`;

exports[`PageSubtitle - Snapshots > matches snapshots for different configurations 3`] = `
<h2
  class="text-xl lg:text-2xl text-muted-foreground font-light leading-normal"
>
  Title prop subtitle
</h2>
`;
